src/Model/CachingIterator.php

```diff
@@ -61,8 +64,7 @@ class CachingIterator implements Countable, Iterator
      */
     public function __construct(Traversable $traversable)
     {
-        $this->iterator = $this->wrapTraversable($traversable);
-        $this->storeCurrentItem();
+        $this->iterator = new IteratorIterator($traversable);
```

Comment:
Would it make sense to just prepare the iterator here and store the current item? That might allow you to remove the `prepareIterator()` calls from other methods.

As-is, there might be a subtle behavioral change, since the original implementation did rewind the traversable, create the generator, and iterate once to store the current item?